contracts/RootChain/RootChain.sol

```diff
@@ -109,22 +104,27 @@ contract RootChain {
         payable
     {
         var txList = txBytes.toRLPItem().toList();
-        require(txList.length == 11);
-        for (uint256 i; i < 6; i++) {
-            require(txList[i].toUint() == 0);
-        }
+        require(txList[6].toAddress() == msg.sender);
```

Comment:
Not sure if this is necessary? If you prefer payment on a specific Plasma chain, i could pay by depositing to your address from the root chain.

Reply:
If the address is malformed, it will cause toAddress to revert. If the address is valid but not correct, then we've simply moved the typo issue from happening on the root chain to happening on the plasma chain. The benefit is you can catch the error and just not send the confirmSig, probably won't happen if the process is automated in the future.

The benefit of allowing people to send to arbitrary addresses is that I can send to your preferred side chain without being on it myself. There is a lot of hassle to joining a plasma chain just to send one payment (validating, syncing, etc.).

Reply:
The revert part makes sense. If the address is valid but not correct, the money gets lost. the confirmsig needed for withdrawal cannot be generated